Update PostgreSQL_psycopg2.md

PatrickFidler opened the pull request:
I am commiting the first version of my A2 contribution. I reworked the PostgreSQL and psycopg2 section to make it more clear on how psycopg2 worked, including how to install/run it and how to use it effectively.

colewiltse reviewed (changes requested):
The changes you have made make it much clearer to understand how to set up and use psycopg2. Specifically, the in-depth steps you updated clearly state what you are doing and why, allowing the user to understand without just following blind instructions.

I do however have some small requests. Near the end of the document you briefly mention using `fetchone()` and `fetchmany(x)`. It would be helpful to provide a real-world scenario to show why you would want to use these functions. Also, for the links you provided in the additional resources section, a short description of what these links are may be helpful to give the reader a taste of what to expect.

PatrickFidler commented:
Great suggestions.
- Added practical example to `PostgreSQL_psycopg2.md` showing why functions like `fetchone()` are useful
- Added descriptors to both links in the Additional Resources section to add clarity about their inclusion
